### ISSUE TITLE
Fix docs for directly calling Operations from the client

### ISSUE THREAD

**Issue by sodic** [wasp-lang/wasp#1923]
This example is missing a mention of passing the `queryCacheKey`: https://wasp-lang.dev/docs/data-model/operations/queries#using-queries.

This example is missing a cast necessary for accessing the `queryCacheKey` in TypeScript: https://wasp-lang.dev/docs/data-model/operations/queries#using-queries.

Decide on the following points and update the code/docs accordingly:
 - Do we want to extract the `queryCacheKey` when inside the Query and spare the user from having to pass it in themselves?
 - Do we want to document the existence of the `queryCacheKey` in the type we expose to the user (instead of hiding it like we do now)?

**Comment by Martinsos** [comment]
Relevant discord convo: https://discord.com/channels/686873244791210014/1222460717136085054/1222460717136085054 .

**Comment by sodic** [comment]
## 1 The problem

Our API (as of 0.13.2) our API for directly calling Queries looks like this:
```typescript
import { getFoo } from 'wasp/client/operations'

const foo = await getFoo((getFoo as any).queryCacheKey, payload)
```
This sucks for three reasons:
1. Users are forced to extract the `queryCacheKey` from a Query function... only to pass it into that same function. It's redundant and a little silly.
1. `getFoo`'s public type isn't aware that `getFoo` has a field called `queryCacheKey`, which forces to user to assert it `as any`.
1.  Finally, this machinery isn't properly documented :upside_down_face: 

## 2 Proposed solution
Let's allow our user to do this:
```typescript
import { getFoo } from 'wasp/client/operations'

const foo = await getFoo(payload)
```
It's much simpler, and **I can't think of a use case where they'd want to call a Query without using its cache** (or using another Query's cache).

> [!IMPORTANT]
> @Martinsos @infomiho The bold part is the big assumption behind this change. If you can imagine a circumstance where our users need this level of control, tell me and I'll rethink the API.
>
> Also won't lose the ability to access Query's `queryCacheKeys`. They just won't be required to.

## 3 Implementing the proposed solution
Instead of receiving the cache key from the user, the query function would read it from its lexical scope (a closure).

This can be done without resorting to deprecated features such as `arguments.callee`: https://stackoverflow.com/a/4654426

> [!CAUTION]
> This would be a breaking change for
>   -  Directly calling Queries on the client - Query functions lose their first positional argument.
>   - `useQuery` - The type would change, but users probably wouldn't notice since nothing would really break (users pass in Queries, which would now have a different type).

**Comment by infomiho** [comment]
Sounds good!

I'd maybe consider doing this in parallel with us bumping React Query to v5 to minimise the number of times we do a breaking change :)

**Comment by sodic** [comment]
> I'd maybe consider doing this in parallel with us bumping React Query to v5 to minimise the number of times we do a breaking change :)

Good idea, but the sprint includes this task and doesn't include updating React Query.

We could maybe do that one in the next sprint and then release them together? Or perhaps the breaking changes aren't that big of a deal.

@Martinsos What do the sprint gods say?

**Comment by Martinsos** [comment]
Breaking changes are not a big deal, but if we can save time on testing and work needed to pump out a new release, then that is a good reason! And we don't have to make a release after every sprint. So I am generally ok with grouping changes, if there is nothing urgent we want to release. At least so the sprint gods tell me :P.

API you proposed makes sense to me @sodic . I wonder what the idea was before, why didn't we do it like this at the very first -> was it because we just didn't put effort to achieve this level of DX, or was it because we had some other use cases in mind. I can't think of any right now though.

There is an alternative also: `const foo = await getFoo.call(payload)`. Meaning that we could have this `.call()` function implemented as a member of `getFoo`, and it would pass query cache key to `getFoo`. That means `getFoo` can remain as it is now, flexible to be called with any query cache key.
On the other hand, we if we stick with approach you proposed, this is also what we can do to avoid customizing cache key if needed: `const foo = await getFoo.withCacheKey(cacheKey, payload)`.

That said, I think I like the idea you proposed @sodic and we should go with it for now. We can adapt if new use cases appear.
